epochStart/metachain/rewards.go

```diff
@@ -391,245 +170,14 @@ func (rc *rewardsCreator) VerifyRewardsMiniBlocks(metaBlock *block.MetaBlock, va
-	}
-}
-
-// DeleteTxsFromStorage deletes data from storage
-func (rc *rewardsCreator) DeleteTxsFromStorage(metaBlock *block.MetaBlock, body *block.Body) {
-	if check.IfNil(metaBlock) || check.IfNil(body) {
-		return
-	}
-
-	for _, miniBlock := range body.MiniBlocks {
-		if miniBlock.Type != block.RewardsBlock {
-			continue
-		}
-
-		for _, txHash := range miniBlock.TxHashes {
-			_ = rc.rewardsStorage.Remove(txHash)
-		}
-	}
-
-	for _, mbHeader := range metaBlock.MiniBlockHeaders {
-		if mbHeader.Type == block.RewardsBlock {
-			_ = rc.miniBlockStorage.Remove(mbHeader.Hash)
-		}
-	}
+	return rc.verifyCreatedRewardMiniBlocksWithMetaBlock(metaBlock, createdMiniBlocks)
 }
 
 // IsInterfaceNil return true if underlying object is nil
 func (rc *rewardsCreator) IsInterfaceNil() bool {
 	return rc == nil
 }
 
-func createBroadcastTopic(shardC sharding.Coordinator, destShId uint32) string {
-	transactionTopic := factory.RewardsTransactionTopic +
-		shardC.CommunicationIdentifier(destShId)
-	return transactionTopic
-}
-
-// RemoveBlockDataFromPools removes block info from pools
-func (rc *rewardsCreator) RemoveBlockDataFromPools(metaBlock *block.MetaBlock, body *block.Body) {
-	if check.IfNil(metaBlock) || check.IfNil(body) {
-		return
-	}
-
-	transactionsPool := rc.dataPool.Transactions()
-	miniBlocksPool := rc.dataPool.MiniBlocks()
-
-	for _, miniBlock := range body.MiniBlocks {
-		if miniBlock.Type != block.RewardsBlock {
-			continue
-		}
-
-		strCache := process.ShardCacherIdentifier(miniBlock.SenderShardID, miniBlock.ReceiverShardID)
-		transactionsPool.RemoveSetOfDataFromPool(miniBlock.TxHashes, strCache)
-	}
-
-	for _, mbHeader := range metaBlock.MiniBlockHeaders {
-		if mbHeader.Type != block.RewardsBlock {
-			continue
-		}
-
-		miniBlocksPool.Remove(mbHeader.Hash)
-
-		log.Trace("RemoveBlockDataFromPools",
-			"hash", mbHeader.Hash,
-			"type", mbHeader.Type,
-			"sender", mbHeader.SenderShardID,
-			"receiver", mbHeader.ReceiverShardID,
-			"num txs", mbHeader.TxCount)
-	}
-}
-
-//TODO evaluate if this function will remain here or be moved inside a new rewardsCreator implementation
-// in case it should remain here, we need a softfork protection for this call
-func (rc *rewardsCreator) prepareStakingData(validatorsInfo map[uint32][]*state.ValidatorInfo) error {
-	sw := core.NewStopWatch()
-	sw.Start("prepareRewardsFromStakingSC")
-	defer func() {
-		sw.Stop("prepareRewardsFromStakingSC")
-		log.Debug("rewardsCreator.prepareRewardsFromStakingSC time measurements", sw.GetMeasurements())
-	}()
-
-	for _, validatorInfoSlice := range validatorsInfo {
-		for _, validatorInfo := range validatorInfoSlice {
-			err := rc.stakingDataProvider.PrepareDataForBlsKey(validatorInfo.PublicKey)
-			if err != nil {
-				//TODO uncomment this return when this function will be used
-				//return err
-			}
-		}
-	}
-
-	return nil
+func (rc *rewardsCreator) isRewardsFix1Enabled(epoch uint32) bool {
+	return epoch > rc.rewardsFix1EnableEpoch
```

Comment:
">" because fix is needed for every epoch after the enabled epoch. 
not ">=" because for economics it would affect the previous epoch (e.g if we set enable epoch 10,  with ">=" it would affect the rewards computation for epoch 9).